src/tribler/gui/tests/test_tribler_window.py

```diff
@@ -0,0 +1,29 @@
+from unittest.mock import Mock, patch
+
+import pytest
+from PyQt5.QtWidgets import QFileDialog
+
+from tribler.gui.tribler_window import TriblerWindow
+
+
+# pylint: disable=redefined-outer-name
+@pytest.fixture
```

Comment:
Suggestion to adhere to our Pylint rules:

```suggestion
@pytest.fixture(name="tribler_window")
def fixture_tribler_window():
```

That aside: please make sure the tests pass and re-request my review.

Reply:
Thank you for the suggestion. From my point of view, for this particular case (for `test_*.py` files), it is preferable to use the following construction:

**Example 1**
```python
# pylint: disable=redefined-outer-name
@pytest.fixture
def fixture_name():
    ...
``` 

Instead of the suggestion by pylint:

**Example 2**
```python
@pytest.fixture(name="fixture_name")
def fixture_fixture_name():
    ...
```

Pylint is aimed at helping developers but not to be blindly obeyed. In many cases, pylint does a great job. However, in this particular case (from my point of view), the suggestion doesn't protect us from any error and does not make the code better. Instead, it makes the code more polluted as it requires unnecessary syntax complication.

I tried both approaches with Tribler, and based on my experience, **Example 1** is as correct as **Example 2**, but it looks better and requires less time to write and correct, which saves developers' time.

Reply:
Sure. Consider these two examples. First, by not specifying a name, everything seems fine:

![case1](https://github.com/Tribler/tribler/assets/3630389/69df2b8a-d733-41ae-9378-112bef6779ed)

Then, option two, we expose a problem with our fixtures by using the `name=` syntax:

![case2](https://github.com/Tribler/tribler/assets/3630389/a93a6fc9-fcaf-42df-b195-0d233b1575b3)

Because the reference no longer exists, we see that there is something wrong with our fixture test.

Reply:
> However, the error with the "something" fixture in your first example will pop up with the first test run.

You misunderstand the core issue. The problem is that **it will not show up** in the test run. Here is the output of the first snippet, nothing pops up:

```
============================= test session starts ==============================
collecting ... collected 2 items

test_something.py::test_2 
test_something.py::test_1 

============================== 2 passed in 0.51s ===============================
PASSED                                         [ 50%]PASSED                                         [100%]
```

Here is the snippet again in more copy-pasteable form, because I can see you didn't run it before making your claim.

```python
from pytest import fixture


@fixture()
def something():
    return True


def test_1(something):
    assert something


def test_2():
    assert something
```

Reply:
Your tests are equivalent to the following:

```python
def test_1(something, port):
    raise TypeError
    assert something.property == 'there is an object normally returns from fixtures'


def test_2():
    raise TypeError
    assert something.property == 'there is an object normally returns from fixtures'
```

You're not even reaching the naming shadowing code.

In fact, your example code is doubly-wrong. Even if I strip out the forced `TypeError`:

```python
def test_1(something, port):
    assert something.property == 'there is an object normally returns from fixtures'


def test_2():
    assert something.property == 'there is an object normally returns from fixtures'
```

You're ignoring the issue of name shadowing by explicitly testing for a property that is supposed to be there.

A more appropriate example is this:

```python
from pytest import fixture
from unittest.mock import Mock


@fixture
def something():
    mock = Mock(return_value='there is an object normally returns from fixtures')
    return mock()


@fixture
def port():
    return 1234  # or sometimes there is indeed the value like a port


def test_1(something, port):
    assert something != 'this is something else'
    assert port != 0


def test_2():
    assert something != 'this is something else'
    assert port != 0
```

I use `!=` instead of a bare `assert` in this case, just to show that **this is an entire class of problems** and not just some fringe case with bare asserts. You can really make this check as complex as you want.

Reply:
Let me add this: I'll stop here. I don't think I can be convinced of your implementation.

That said, if you still insist that we should ignore the pytest docs, Pylint (+ Codacy) warnings, @kozlovsky's concerns and my own and that the examples I gave are not relevant in practice: we can assign a different reviewer to this PR.

Reply:
I appreciate your effort to analyze my examples. Although I don't agree with the conclusions, I'm not going to try to change your mind because this particular part is not that important to the general topic at hand.

The funny thing is that we both actually started the discussion about shadowing without having an example of real shadowing. We started from here:

```python
from pytest import fixture


@fixture()
def something():
    return True


def test_1(something):
    assert something


def test_2():
    assert something
```

This example does not have anything related to the shadowing. I haven't found any definition for shadowing in general but there's a definition for variable shadowing https://en.wikipedia.org/wiki/Variable_shadowing

In the case of fixtures there is a "function name" shadowing:

> If a fixture is used in the same module in which it is defined, the function name of the fixture will be shadowed by the function arg that requests the fixture

I understand it as 

```python
from pytest import fixture


@fixture()
def something():
    return True


def test_1(something):  # <- shadowing happens here and this fact could lead only to harmless warnings.
    assert something
```

I can also provide an example of real shadowing, although it does not relate to the original "function name" shadowing from the pytest documentation:

```python
from pytest import fixture


@fixture()
def something():
    return True


def something():
    return False


def test_1(something):
    assert something

```

Summarising my point: you're right that there is shadowing correctly reported by pylint, but it can be safely ignored in this particular code (that I wrote for this PR). 

From my point of view (which could be wrong), that warning can be ignored pretty much everywhere in Tribler. Anyway, the question of whether we should ignore the warning for the entire test suite should be raised in a dev meeting, as it applies to everyone.

For a "controversial" PR like this, it's good to learn how we can move forward and reach a consensus, even if not all parties are fully satisfied with the final decision.

The best-written guide for the review process, to my knowledge, is the "Google Engineering Practices Documentation." So, we can use it as a starting point to make a decision on whether a PR is good enough to be accepted or not:
https://github.com/google/eng-practices/blob/master/review/reviewer/index.md

Reply:
I guess I can provide the "function name shadowing" problem example:

```python
from pytest import fixture


@fixture()
def something():
    return True


def test_1(something):
    assert something

    # here by mistake a developer could call the "something" function but here is the fixture, eg. the function result
    result = something()
```

Reply:
From your comment I deduce that you do not see the point I'm trying to make. The issue with name shadowing is not `test_1` but it is `test_2`. The problem with `test_2` is that it may be introduced by someone else in the future. So, even if `test_1` works right now, someone can get into trouble in the future due to your (avoidable) construction. If you want to relate it to the Google practices, this practice degrades the code quality.

The reason I suggested leaving this to another reviewer is to not make this PR controversial. Perhaps another reviewer can explain this issue in a different way that makes more sense to you. I would like to urge you to allow a second reviewer opinion on this.

Reply:
@egbertbouman @xoriole please, share your opinion 🙏

Reply:
@qstokkink, no-no, I think I understand you. But in test `test_2`, there is no shadowing.

And yes, someone in the future could introduce mistakes (arithmetic, logical, or even related to shadowing) to `test_2`, which has nothing in common with the shadowing mentioned in the original pylint warning.

Reply:
@drew2a Judging from your comment, I still do not think that you do understand. There is **one** specific type of situation that can happen solely due to the construction that causes name shadowing (in `test_1`). Not multiple as you just said. I gave an instance of this type of situation in the example above (https://github.com/Tribler/tribler/pull/7764#discussion_r1422721124).

The shadowing warning you get for `test_1` causes trouble for `test_2`, which will not get a warning. It (`test_2`) will pass, even though it can and should fail.

Reply:
@qstokkink, with all due respect, it's not shadowing that causes the potential problem; it's just a badly written test that caused the problem.

Here's a slightly changed example that "has problems due to shadowing." It still has them, even without any shadowing:

```python
def something():
    return True


def test_2():
    assert something
```

Reply:
@drew2a Yes, thank you, that is indeed the core issue: if `something` is now made into a `fixture`, which uses a `name` argument, exactly this type of error cannot happen.